src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -1391,57 +1392,318 @@ static final Color hexToColor(String value) {
+        else if (str.equalsIgnoreCase("Palevioletred"))
+            color = hexToColor("#db7093");
+        else if (str.equalsIgnoreCase("Papayawhip"))
+            color = hexToColor("#ffefd5");
+        else if (str.equalsIgnoreCase("Peachpuff"))
+            color = hexToColor("#ffdab9");
+        else if (str.equalsIgnoreCase("Peru"))
+            color = hexToColor("#cd853f");
+        else if (str.equalsIgnoreCase("Pink"))
+            color = hexToColor("#ffc0cb");
+        else if (str.equalsIgnoreCase("Plum"))
+            color = hexToColor("#dda0dd");
+        else if (str.equalsIgnoreCase("Powderblue"))
+            color = hexToColor("#b0e0e6");
+        else if (str.equalsIgnoreCase("Rebeccapurple"))
+            color = hexToColor("#663399");
+        else if (str.equalsIgnoreCase("Rosybrown"))
+            color = hexToColor("#bc8f8f");
+        else if (str.equalsIgnoreCase("Royalblue"))
+            color = hexToColor("#4169e1");
+        else if (str.equalsIgnoreCase("Saddlebrown"))
+            color = hexToColor("#8b4513");
+        else if (str.equalsIgnoreCase("Salmon"))
+            color = hexToColor("#fa8072");
+        else if (str.equalsIgnoreCase("Sandybrown"))
+            color = hexToColor("#f4a460");
+        else if (str.equalsIgnoreCase("Seagreen"))
+            color = hexToColor("#2e8b57");
+        else if (str.equalsIgnoreCase("Seashell"))
+            color = hexToColor("#fff5ee");
+        else if (str.equalsIgnoreCase("Sienna"))
+            color = hexToColor("#a0522d");
+        else if (str.equalsIgnoreCase("Skyblue"))
+            color = hexToColor("#87ceeb");
+        else if (str.equalsIgnoreCase("Slateblue"))
+            color = hexToColor("#6a5acd");
+        else if (str.equalsIgnoreCase("Slategray"))
+            color = hexToColor("#708090");
+        else if (str.equalsIgnoreCase("Slategrey"))
+            color = hexToColor("#708090");
+        else if (str.equalsIgnoreCase("Snow"))
+            color = hexToColor("#fffafa");
+        else if (str.equalsIgnoreCase("Springgreen"))
+            color = hexToColor("#00ff7f");
+        else if (str.equalsIgnoreCase("Steelblue"))
+            color = hexToColor("#4682b4");
+        else if (str.equalsIgnoreCase("Tan"))
+            color = hexToColor("#d2b48c");
+        else if (str.equalsIgnoreCase("Thistle"))
+            color = hexToColor("#d8bfd8");
+        else if (str.equalsIgnoreCase("Tomato"))
+            color = hexToColor("#ff6347");
+        else if (str.equalsIgnoreCase("Turquoise"))
+            color = hexToColor("#40e0d0");
+        else if (str.equalsIgnoreCase("Violet"))
+            color = hexToColor("#ee82ee");
+        else if (str.equalsIgnoreCase("Wheat"))
+            color = hexToColor("#f5deb3");
+        else if (str.equalsIgnoreCase("Whitesmoke"))
+            color = hexToColor("#f5f5f5");
+        else if (str.equalsIgnoreCase("Yellowgreen"))
+            color = hexToColor("#9acd32");
```

Comment:
This should probably use <code>switch([str.toLowerCase][String::toLowerCase]\([Locale.ROOT]))</code> instead of a linear search.

[String::toLowerCase]: https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/String.html#toLowerCase(java.util.Locale)
[Locale.ROOT]: https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/util/Locale.html#ROOT

Reply:
Hi @ExE-Boss, Thanks for reviewing, I made several propositions here https://github.com/scientificware/jdk/issues/12.

Including switch usage with direct Color generating.
The last on is to move this logic to Color.java.

May I have your comments about them ?

Reply:
Hello @ExE-Boss,
A new implementation based on your comments and some performance tests.